internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -394,6 +396,57 @@ func (r *ClusterDeploymentsReconciler) installDay1(ctx context.Context, log logr
 	return r.updateStatus(ctx, log, clusterInstall, cluster, nil)
 }
 
+func (r *ClusterDeploymentsReconciler) spokeKubeClient(ctx context.Context, clusterDeployment *hivev1.ClusterDeployment) (spoke_k8s_client.SpokeK8sClient, error) {
+	adminKubeConfigSecretName := getClusterDeploymentAdminKubeConfigSecretName(clusterDeployment)
+
+	namespacedName := types.NamespacedName{
+		Namespace: clusterDeployment.Namespace,
+		Name:      adminKubeConfigSecretName,
+	}
+
+	secret, err := getSecret(ctx, r.Client, r.APIReader, namespacedName)
+	if err != nil {
+		r.Log.WithError(err).Errorf("failed to get kubeconfig secret %s", namespacedName)
+		return nil, err
+	}
+	if err = ensureSecretIsLabelled(ctx, r.Client, secret, namespacedName); err != nil {
+		r.Log.WithError(err).Errorf("failed to label kubeconfig secret %s", namespacedName)
+		return nil, err
+	}
+	return r.SpokeK8sClientFactory.CreateFromSecret(secret)
+}
+
+func (r *ClusterDeploymentsReconciler) updateWorkerMcpPaused(ctx context.Context, log logrus.FieldLogger, clusterInstall *hiveext.AgentClusterInstall, clusterDeployment *hivev1.ClusterDeployment) error {
+	agents, err := findAgentsByAgentClusterInstall(r.Client, ctx, log, clusterInstall)
+	if err != nil {
+		log.WithError(err).Errorf("failed to find agents for cluster install %s/%s", clusterInstall.Namespace, clusterInstall.Name)
+		return err
+	}
+	var paused, shouldUpdate bool
+	funk.ForEach(agents, func(agent *aiv1beta1.Agent) {
+		if !funk.IsEmpty(agent.Spec.NodeLabels) && agent.Spec.MachineConfigPool != "" && agent.Status.DebugInfo.State != models.HostStatusInstalled {
```

Comment:
who is moving the agent to installed?

Reply:
I know we talked about it but seems it is not clear not only for me, maybe better to get node from db in that case and verify it is day2?

Reply:
Or actually we should have cluster object here why not just to go through its nodes?

Reply:
if `agent.Status.DebugInfo.State != models.HostStatusInstalled` cover day1 hosts then a comment should be enough, i was thinking that it's a day2 host that is installed and this is what confused me.

Reply:
Added comments